lib/compiler/src/beam_asm.erl

```diff
@@ -68,7 +68,18 @@ assemble({Mod,Exp0,Attr0,Asm0,NumLabels}, ExtraChunks, SourceFile, Opts, Compile
     {Asm,Attr} = on_load(Asm0, Attr0),
     Exp = cerl_sets:from_list(Exp0),
     {Code,Dict2} = assemble_1(Asm, Exp, Dict1, []),
-    build_file(Code, Attr, Dict2, NumLabels, NumFuncs, ExtraChunks, SourceFile, Opts, CompilerOpts).
+
+    %% Populate atom dict with atoms that were optimized away during Core Fold
+    %% Atoms may possibly exist in atom Dict, but we attempt to add them anyway
+    OptAtoms = proplists:get_value(optimized_away_atoms, Attr0, []),
```

Comment:
I would prefer a name that can't possible be in conflict with a user-defined attribute. Perhaps you could a binary as a key?

`OptAtoms = proplists:get_value(<<"optimized_away_atoms">>, Attr0, []),

If that would be problematic, perhaps you could add an underscore like this:

`OptAtoms = proplists:get_value(_optimized_away_atoms, Attr0, []),